contracts/mocks/DCAHubCompanion/DCAHubCompanionHubProxyHandler.sol

```diff
@@ -3,4 +3,49 @@ pragma solidity >=0.8.7 <0.9.0;
 
 import '../../DCAHubCompanion/DCAHubCompanionHubProxyHandler.sol';
 
-contract DCAHubCompanionHubProxyHandlerMock is DCAHubCompanionHubProxyHandler {}
+contract DCAHubCompanionHubProxyHandlerMock is DCAHubCompanionHubProxyHandler {
+  struct DepositProxyCall {
+    IDCAHub hub;
+    address from;
+    address to;
+    uint256 amount;
+    uint32 amountOfSwaps;
+    uint32 swapInterval;
+    address owner;
+    IDCAPermissionManager.PermissionSet[] permissions;
+    bytes miscellaneous;
+  }
+
+  DepositProxyCall[] private _depositProxyCalls;
+
+  function depositProxyCalls() external view returns (DepositProxyCall[] memory) {
+    return _depositProxyCalls;
+  }
+
+  function depositProxy(
+    IDCAHub _hub,
+    address _from,
+    address _to,
+    uint256 _amount,
+    uint32 _amountOfSwaps,
+    uint32 _swapInterval,
+    address _owner,
+    IDCAPermissionManager.PermissionSet[] calldata _permissions,
+    bytes calldata _miscellaneous
+  ) public payable override returns (uint256 _positionId) {
+    _depositProxyCalls.push();
+    DepositProxyCall storage _ref = _depositProxyCalls[_depositProxyCalls.length - 1];
```

Comment:
Is there a reason why you split it like this?

Reply:
+++ I wanna know! 😂

Reply:
Hahah. I tried to push it directly, but Solidity doesn't support array conversions from calldata to storage yet, so I had to do it manually